pallets/gear/src/manager/mod.rs

```diff
@@ -404,42 +403,27 @@ where
         ProgramStorageOf::<T>::waiting_init_remove(program_id);
     }
 
-    fn process_failed_init(program_id: ProgramId, origin: ProgramId, executed: bool) {
+    fn process_failed_init(program_id: ProgramId, origin: ProgramId) {
         // Some messages addressed to the program could be processed
         // in the queue before init message. For example, that could
         // happen when init message had more gas limit then rest block
         // gas allowance, but a dispatch message to the program was
         // dequeued. The other case is async init.
         Self::clean_waitlist(program_id);
 
-        ProgramStorageOf::<T>::update_program_if_active(program_id, |p, bn| {
+        let _ = ProgramStorageOf::<T>::update_program_if_active(program_id, |p, bn| {
```

Comment:
you can invent new invariants infinitely. if you use some function param only for panics and also create an additional field for in some structures, than it's bad solution, because developer while reading the code suppose that it is some necessary for logic param.

Also `executed` is a bad name and sometimes `executed` has `true` value even in cases nothing was executed.

Reply:
Yes you can, and you must make sure that each of your brand-new ideas fit in the runtime logic

Reply:
At least, put log::error, but I still think the panic there is better if we didn't broke anything on refactor

Reply:
as I have already tell `executed` is not really executed it is just set to true or false in order to fit this panic cases, so this panics are useless